### PR TITLE
Don't move from contained value of optional<const T>

### LINKED ISSUE #2458: `<optional>`: `std::optional` move constructor moves a const object
**Describe the bug**
A simple example below shows that the move constructor of the contained object is invoked when an std::optional<const TYPE> is move-constructed from another instance of std::optional<const TYPE>. This behavior is (1) surprising and (2) different from that of libc++/libstdc++.

https://godbolt.org/z/5qxT4v8xd

**Command-line test case**
```
C:\Temp>type main.cpp
#include <iostream>
#include <utility>
#include <optional>

struct Copyable {
    Copyable() = default;
    Copyable(const Copyable&) {
        std::cout << "Copyable copied\n";
    }
    Copyable(Copyable&&) {
        std::cout << "Copyable moved\n";
    }
};

template <typename TYPE>
void testMove() {
    std::optional<TYPE> orig;
    orig.emplace();
    std::optional<TYPE> moved = std::move(orig);
}

int main() {
    testMove<Copyable>();
    testMove<const Copyable>();
}

C:\Temp>cl /EHsc /W4 /WX /std:c++17 .\main.cpp
Microsoft (R) C/C++ Optimizing Compiler Version 19.31.30818 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

main.cpp
Microsoft (R) Incremental Linker Version 14.31.30818.0
Copyright (C) Microsoft Corporation.  All rights reserved.

/out:main.exe
main.obj

C:\Temp>.\main.exe
Copyable moved
Copyable moved
```

**Expected behavior**
Copyable moved
Copyable copied

**STL version**
    Microsoft Visual Studio Community 2022 Preview
    Version 17.1.0 Preview 1.1

**Additional context**
 DevCom-1628644

### PR DESCRIPTION
**Pull request by CaseyCarter**
...unless `T` supports it, of course.

Fixes #2458
Fixes DevCom-1628644 / VSO-1458298 / AB#1458298

### CODE REVIEW

**Review by ddr533 — APPROVED** [review]

**Review by fsb4000 — APPROVED** [review]
Note to self: that issue was because `_Value` defined as `remove_const_t<_Ty> _Value;` 
So we always could move `_Value`. and now with `operator*` we take into account the `const` of `_Ty`.

File: tests/std/tests/P0220R1_optional/test.cpp [hunk 1]
```diff
@@ -7382,6 +7382,47 @@ namespace msvc {
...
+            ConstMovable() = default;
+            ConstMovable(const ConstMovable&) {
+                last = action::copy;
+            }
+            ConstMovable(const ConstMovable&&) {
+                last = action::move;
+            }
+        };
+
+        template <class T, action result>
```
Comment (StephanTLavavej) [inline tests/std/tests/P0220R1_optional/test.cpp:7412]: Style: We conventionally capitalize template parameters, so this should be `Result`.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Looks good to meow! :smiley_cat: I went ahead and pushed a trivial name change, plus a merge with `main` since it had been over a month (and we merged your significant LLVM update).

**Review by barcharcraz — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for fixing this moving bug! :bug: :lady_beetle: :smile_cat:

**Pull request merged**